src/parcsr_ls/HYPRE_parcsr_ls.h

```diff
@@ -1228,10 +1247,34 @@ HYPRE_Int HYPRE_BoomerAMGSetDSLUThreshold (HYPRE_Solver solver,
  * @param cpt_coarse_index [IN] indexes of C points to be kept
  **/
 HYPRE_Int HYPRE_BoomerAMGSetCpointsToKeep(HYPRE_Solver solver,
-				HYPRE_Int  cpt_coarse_level,
-				HYPRE_Int  num_cpt_coarse,
-				HYPRE_Int *cpt_coarse_index);
-/*
+                                          HYPRE_Int    cpt_coarse_level,
+                                          HYPRE_Int    num_cpt_coarse,
+                                          HYPRE_Int   *cpt_coarse_index);
+
+/**
+ * (Optional) Set fine points in the first level.
+ *
+ * @param solver [IN] solver or preconditioner
+ * @param num_fpt [IN] number of fine points
+ * @param fpt_index [IN] global indices of fine points
+ **/
+HYPRE_Int HYPRE_BoomerAMGSetFPoints(HYPRE_Solver solver,
```

Comment:
Should this routine have a similar name to HYPRE_BoomerAMGSetCpointsToKeep()?  I actually think that "ToKeep" is somewhat redundant, but if we remove it we may cause backward compatibility issues for some users (probably not many though).  I would be interested in hearing opinions here.

Reply:
@rfalgout, the interface is a bit different between the two, although the idea is similar. One sets F points *only* for the finest level, whereas the other keeps specific points as C points until some coarse level is reached. So the notion of "keep"-ing the points is not too bad (for MGR, I used MGRSetReservedCoarseNodes). That said, I don't think SetFPoints needs "ToKeep" to be descriptive. I also don't think many external users are using the SetCpointsToKeep() routine. I use it in MGR, but it should be an easy change if we switch the name.

Reply:
Hi all, I chose to drop "ToKeep" for the reason that Daniel pointed out.

Reply:
Thanks @oseikuffuor1 and @victorapm .  If we do anything here, I would vote for removing "ToKeep" from the SetCPoints function.  But, I don't have an issue with leaving it the way it is currently either.

Reply:
We could provide both functions with and without ToKeep, but deprecate the ToKeep one and hide it.